### ISSUE TITLE
MixStyle.py can't be used normally when batchsize is an odd number

### ISSUE THREAD

**Issue by EvenTiming** [Feaxure-fresh/TL-Bearing-Fault-Diagnosis#6]
When I use the code of MixStyle.py to train my model. I found that it can not train normally when my batchsize is an oddnumber.
And I find the problem maybe caused by this code 
` perm = torch.arange(batch_size - 1, -1, -1)  # inverse index
        perm_b, perm_a = perm.chunk(2)
        perm_b = perm_b[torch.randperm(batch_size // 2)]
        perm_a = perm_a[torch.randperm(batch_size // 2)]
        perm = torch.cat([perm_b, perm_a], 0)`
and after my change:
` perm = torch.arange(batch_size - 1, -1, -1)  # inverse index
        perm_b, perm_a = perm.chunk(2)
        perm_b = perm_b[torch.randperm((batch_size + 1) // 2)]
        perm_a = perm_a[torch.randperm(batch_size // 2)]
        perm = torch.cat([perm_b, perm_a], 0)`
it can work normally

**Comment by Feaxure-fresh** [comment]
The MixStyle method, which involves shuffling and mixing style features between different samples in a batch during training, typically requires an even batch size for a specific reason related to its implementation. Please refer to their paper on MixStyle method.
The requirement of an even batch size in the MixStyle method is primarily to ensure that every sample in a batch can be paired with another for effective style mixing. It is not recommended to use odd batch sizes when training models using the MixStyle method. If you have to use an odd batch size, please modify the relevant code (the code provided in your issue) yourself to make it work properly and comply with its theoretical basis.

**Comment by EvenTiming** [comment]
Thank you for your patient reply, I learned a lot from your reply. God bless you !
